pytorch_lightning/strategies/ddp2.py

```diff
@@ -73,3 +74,11 @@ def set_world_ranks(self) -> None:
             return
         self.cluster_environment.set_global_rank(self.node_rank)
         self.cluster_environment.set_world_size(self.num_nodes)
+
+    @classmethod
+    def register_strategies(cls, strategy_registry: Dict) -> None:
+        strategy_registry.register(
+            cls.strategy_name,
+            cls,
+            description=f"{cls.__class__.__name__}",
+        )
```

Comment:
Yeah, I think it won't work. I have added into #11449 follow up item 3. If we have it in base class, the cls will be the base class instead of the class inheritance the base class.